lib/internal/per_context/primordials.js

```diff
@@ -489,6 +536,23 @@ primordials.SafePromiseAllSettled = (promises, mapFn) =>
     SafePromise.allSettled(arrayToSafePromiseIterable(promises, mapFn)).then(a, b)
   );
 
+/**
+ * Should only be used when we only care about waiting for all the promises to
+ * settle, not what value they resolve or reject to.
+ * @param {Promise<any>[]} promises
+ * @param {(v: Promise<any>, k: number) => Promise<any>} [mapFn]
+ * @returns {Promise<void>}
+ */
+primordials.SafePromiseAllSettledReturnVoid = async (promises, mapFn) => {
+  for (let i = 0; i < promises.length; i++) {
+    try {
+      await (mapFn != null ? mapFn(promises[i], i) : promises[i]);
```

Comment:
```suggestion
      await (mapFn != null ? mapFn(await promises[i], i) : promises[i]);
```

Reply:
Nah, `mapFn` itself can await the value if needed, I don't think it's needed.

Reply:
in the spec one, a mapper never can receive a thenable value. that seems like an unfortunate caveat to keep when it's avoidable?

Reply:
I don't think there's any reason a mapper function would not receive a thenable, what do you mean?
```js
[Promise.resolve()].map(val => assert('then' in val))
```

Reply:
In that example yes, but in a Promise.all call, you'd generally get a promise for an array, and then map over *that*. - it seems kind of useless to provide the mapper function if the caller can `.map` on the array they're providing?

Reply:
That doesn't seem the case in the Node.js codebase, the most common usecase is to have an array of non-Promise values and pass it to an async function AFAICT.

Reply:
alrighty, fair enough since it doesn't have to exactly match the spec :-) feel free to resolve this and https://github.com/nodejs/node/pull/45175#discussion_r1004842540, then.

Reply:
OK, we can always reconsider this if the use-case arises in the future, that would also let us measure what would be the perf implications to add an `await` here.